### PR TITLE
ci: pin typing_extensions

### PR DESCRIPTION
**Pull request by mabdinur**
̶T̶h̶e̶ ̶l̶a̶t̶e̶s̶t̶ ̶r̶e̶l̶e̶a̶s̶e̶ ̶o̶f̶ ̶[̶t̶y̶p̶i̶n̶g̶ ̶e̶x̶t̶e̶n̶s̶i̶o̶n̶s̶]̶(̶h̶t̶t̶p̶s̶:̶/̶/̶g̶i̶t̶h̶u̶b̶.̶c̶o̶m̶/̶p̶y̶t̶h̶o̶n̶/̶t̶y̶p̶i̶n̶g̶_̶e̶x̶t̶e̶n̶s̶i̶o̶n̶s̶/̶b̶l̶o̶b̶/̶m̶a̶i̶n̶/̶C̶H̶A̶N̶G̶E̶L̶O̶G̶.̶m̶d̶#̶r̶e̶l̶e̶a̶s̶e̶-̶4̶6̶0̶-̶m̶a̶y̶-̶2̶2̶-̶2̶0̶2̶3̶)̶ ̶b̶r̶o̶k̶e̶ ̶t̶h̶e̶ ̶d̶e̶b̶u̶g̶g̶e̶r̶,̶ ̶a̶n̶d̶ ̶c̶h̶e̶r̶r̶y̶p̶y̶ ̶t̶e̶s̶t̶ ̶s̶u̶i̶t̶e̶s̶.̶ ̶

̶T̶h̶i̶s̶ ̶P̶R̶ ̶p̶i̶n̶s̶ ̶t̶h̶e̶ ̶v̶e̶r̶s̶i̶o̶n̶ ̶o̶f̶ ̶t̶y̶p̶i̶n̶g̶_̶e̶x̶t̶e̶n̶s̶i̶o̶n̶s̶ ̶u̶s̶e̶d̶ ̶i̶n̶ ̶t̶e̶s̶t̶s̶ ̶t̶o̶ ̶u̶n̶b̶l̶o̶c̶k̶ ̶c̶i̶.̶ ̶T̶h̶i̶s̶ ̶a̶ ̶t̶e̶m̶p̶o̶r̶a̶r̶y̶ ̶m̶i̶t̶i̶g̶a̶t̶i̶o̶n̶.̶ ̶W̶e̶ ̶w̶i̶l̶l̶ ̶n̶e̶e̶d̶ ̶t̶o̶ ̶i̶n̶v̶e̶s̶t̶i̶g̶a̶t̶e̶ ̶t̶h̶e̶ ̶u̶n̶d̶e̶r̶l̶y̶i̶n̶g̶ ̶i̶n̶c̶o̶m̶p̶a̶t̶i̶b̶i̶l̶i̶t̶y̶.̶ ̶

̶F̶a̶i̶l̶e̶d̶ ̶P̶i̶p̶e̶l̶i̶n̶e̶:̶ ̶h̶t̶t̶p̶s̶:̶/̶/̶a̶p̶p̶.̶c̶i̶r̶c̶l̶e̶c̶i̶.̶c̶o̶m̶/̶p̶i̶p̶e̶l̶i̶n̶e̶s̶/̶g̶i̶t̶h̶u̶b̶/̶D̶a̶t̶a̶D̶o̶g̶/̶d̶d̶-̶t̶r̶a̶c̶e̶-̶p̶y̶/̶3̶6̶2̶9̶0̶/̶w̶o̶r̶k̶f̶l̶o̶w̶s̶/̶3̶0̶d̶0̶3̶c̶6̶b̶-̶1̶e̶9̶a̶-̶4̶2̶e̶3̶-̶9̶5̶7̶8̶-̶0̶9̶0̶b̶3̶3̶5̶3̶7̶9̶0̶3̶

___

This PR pins the typing extensions version installled by ddtrace. This is not an acceptable solution, The failing tests or the underlying incompatibility should be documented and/or addressed.
  -  Ex: Ensure riot venvs can override packages installed by ddtrace reqs.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by christophe-papazian — DISMISSED** [review]

**Review by avara1986 — APPROVED** [review]

**Review by christophe-papazian — DISMISSED** [review]

File: setup.py [hunk 1]
```diff
@@ -347,7 +347,7 @@ def get_exts_for(name):
         "cattrs<1.1; python_version<='3.6'",
         "cattrs; python_version>='3.7'",
         "six>=1.12.0",
-        "typing_extensions",
+        "typing_extensions<4.6.0",
```
Comment (Kyle-Verhoog) [inline setup.py:350]: is there an actual incompatibility between ddtrace and `typing_extensions`? I thought the new version of typing_extensions was only affecting a few of our integration packages>

Comment (gnufede) [reply setup.py:350]: Based on the commit message, I'd say not necessarily.

Comment (mabdinur) [reply setup.py]: Yes, this appears to be an issue with just two integrations

**Review by gnufede — DISMISSED** [review]

**Review by emmettbutler — DISMISSED** [review]

**Comment by mabdinur** [comment]
Blocked by: https://github.com/DataDog/riot/pull/209

**Comment by mabdinur** [comment]
Closing in favor of: https://github.com/DataDog/dd-trace-py/pull/5949

**Pull request closed without merging**